src/ts/ssh-tcp/services/portForwardingService.ts

```diff
@@ -239,14 +256,16 @@ export class PortForwardingService extends SshService {
 		}
 
 		remotePort = forwarder.remotePort;
-		const remoteEndPoint = `${remoteIPAddress}:${remotePort}`;
-		this.remoteConnectors.set(remoteEndPoint, forwarder);
+
+		// The remote port is the port sent in the message to the other side,
+		// so the connector is indexed on that port number, rather than the local port.
+		this.remoteConnectors.set(remotePort, forwarder);
 
 		const forwardedPort = new ForwardedPort(localPort, remotePort, false);
 		this.localForwardedPorts.addPort(forwardedPort);
 		forwarder.onDisposed(() => {
 			this.localForwardedPorts.removePort(forwardedPort);
-			this.remoteConnectors.delete(remoteEndPoint);
+			this.remoteConnectors.delete(remotePort);
```

Comment:
Should we check that `remoteConnectors[remotePort]` is still `forwarder`? In case there is a race where new forwarded port uses the same remote port before the old `forwarder` is disposed.

Reply:
There cannot be a race because the SSH request messages are processed sequentially, even if the processing awaits async operations.